src/coreclr/vm/dwbucketmanager.hpp

```diff
@@ -960,11 +960,18 @@ bool BaseBucketParamsManager::GetFileVersionInfoForModule(Module* pModule, USHOR
         // if we failed to get the version info from the native image then fall back to the IL image.
         if (!succeeded)
         {
-            LPCWSTR modulePath = pPEAssembly->GetPath().GetUnicode();
-            if (modulePath != NULL && modulePath != SString::Empty() && SUCCEEDED(DwGetFileVersionInfo(modulePath, major, minor, build, revision)))
+            EX_TRY
             {
-                succeeded = true;
+                SString modulePath{ pPEAssembly->GetPath() };
```

Comment:
This is executing during watson handling. Introducing new allocations in the watson handling is potentially problematic. (I am not sure how much we allocate during watson handling today, but I would expect we try to minimize it.)

I would keep this as is, and maybe just assert that the string we got has `IsNormalized()` set and thus `GetUnicode` won't allocate or cause conversions.